src/python/pants/core/goals/lint.py

```diff
@@ -416,6 +374,113 @@ def _get_error_code(results: Sequence[LintResult]) -> int:
+    ],
+) -> list[LintRequest.SubPartition]:
+    filtered_core_request_types = [
+        request_type for request_type in core_request_types if request_type.name in specified_names
+    ]
+    if not filtered_core_request_types:
+        return []
+
+    core_partition_request_types = {
+        getattr(request_type, "PartitionRequest") for request_type in filtered_core_request_types
+    }
+    target_partitioners = [
+        target_partitioner
+        for target_partitioner in target_partitioners
+        if target_partitioner in core_partition_request_types
+    ]
+    file_partitioners = [
+        file_partitioner
+        for file_partitioner in file_partitioners
+        if file_partitioner in core_partition_request_types
+    ]
+
+    _get_targets = Get(
+        FilteredTargets,
+        Specs,
+        specs if target_partitioners else Specs.empty(),
+    )
+    _get_specs_paths = Get(SpecsPaths, Specs, specs if file_partitioners else Specs.empty())
+
+    targets, specs_paths = await MultiGet(_get_targets, _get_specs_paths)
+
+    def partition_request_get(request_type: type[LintRequest]) -> Get[Partitions]:
+        partition_request_type: type = getattr(request_type, "PartitionRequest")
+        if partition_request_type in target_partitioners:
+            partition_targets_type = cast(LintTargetsRequest, request_type)
+            field_set_type = partition_targets_type.field_set_type
+            field_sets = tuple(
+                field_set_type.create(target)
+                for target in targets
+                if field_set_type.is_applicable(target)
+            )
+            return make_targets_partition_request_get(
+                partition_targets_type.PartitionRequest(field_sets)
+            )
+        else:
+            assert partition_request_type in file_partitioners
+            partition_files_type = cast(LintFilesRequest, request_type)
+            return make_files_partition_request_get(
+                partition_files_type.PartitionRequest(specs_paths.files)
+            )
+
+    all_partitions = await MultiGet(
+        partition_request_get(request_type) for request_type in filtered_core_request_types
+    )
+    partitions_by_request_type = defaultdict(list)
+    for request_type, partition in zip(filtered_core_request_types, all_partitions):
+        partitions_by_request_type[request_type].append(partition)
+
+    def batch(
+        iterable: Iterable[_T], key: Callable[[_T], str] = lambda x: str(x)
+    ) -> Iterator[tuple[_T, ...]]:
+        batches = partition_sequentially(
+            iterable,
+            key=key,
+            size_target=subsystem.batch_size,  # type: ignore[attr-defined]
+            size_max=4 * subsystem.batch_size,  # type: ignore[attr-defined]
+        )
+        for batch in batches:
+            yield tuple(batch)
+
+    lint_batches_by_request_type = {
+        request_type: [
+            (subpartition, key)
+            # @TODO: rename partionss
```

Comment:
Hearing names :smile:

Reply:
`partitions_list` ?

But uh, yea. The only rule of thumb I can imagine is that "partitions" could be things which "need" to be partitioned (because they are incompatible), while "batches" could be things which are split for other reasons. If we applied that, we should probably rename `partition_sequentially` to `batch_sequentially`.

Reply:
> rename partition_sequentially to batch_sequentially.

I'd be +1 on that